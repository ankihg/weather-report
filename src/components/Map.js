import React from 'react';

import {
  ComposableMap,
  Geographies,
  Geography,
  Marker
} from "react-simple-maps";

const geoUrl = "https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/topojson-maps/world-110m.json";

export default class Map extends React.Component {
    render() {
        return (
            <ComposableMap style={{width: '100%', height: '400px', margin: 'auto'}}>
                <Geographies geography={geoUrl}>
                  {({ geographies }) =>
                    geographies.map(geo => <Geography key={geo.rsmKey} geography={geo} />)
                  }
                </Geographies>

                {
                    this.props.city && this.props.coordinates &&
                    <Marker key={this.props.city} coordinates={this.props.coordinates}>
                      <circle r={10} fill="#F00" stroke="#fff" strokeWidth={2} />
                    </Marker>
                }

            </ComposableMap>
        );
    }
}
