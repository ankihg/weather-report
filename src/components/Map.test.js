import React from 'react';
import Map from './Map';
import renderer from 'react-test-renderer';

test('Snapshot of Map component', () => {
    const component = renderer.create(
        <Map city={'Seattle'} coordinates={[-122.33, 47.61]} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
