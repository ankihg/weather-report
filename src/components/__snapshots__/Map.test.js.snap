// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Map component 1`] = `
<svg
  className="rsm-svg "
  style={
    Object {
      "height": "400px",
      "margin": "auto",
      "width": "100%",
    }
  }
  viewBox="0 0 800 600"
>
  <g
    className="rsm-geographies "
  />
  <g
    className="rsm-marker "
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    transform="translate(126.070043393491, 139.98140624321968)"
  >
    <circle
      fill="#F00"
      r={10}
      stroke="#fff"
      strokeWidth={2}
    />
  </g>
</svg>
`;
